pkg/alertmanager/operator.go

```diff
@@ -1040,6 +1044,21 @@ func (c *Operator) selectAlertmanagerConfigs(ctx context.Context, am *monitoring
 	return res, nil
 }
 
+func checkAlertmanagerRootConfig(ctx context.Context, conf *alertmanagerConfig) error {
+	if conf == nil || conf.Route == nil {
+		return nil
+	}
+
+	if len(conf.Route.Matchers) > 0 {
+		return errors.Errorf("'matchers' not permitted on root route: %v", conf.Route.Matchers)
+	}
+
+	if len(conf.Route.MuteTimeIntervals) > 0 {
+		return errors.Errorf("'mute_time_intervals' not permitted on root route: %v", conf.Route.MuteTimeIntervals)
+	}
```

Comment:
another option would be to unset the 2 fields?

Reply:
I thought about that also, we could also do that in sanitise func I think, but I preferred this solution, since I thought it was a large enough issue to force a hard error. I am not tied to it though and can strip it if that is preferable

Reply:
lets keep it this way. I think that fixing bad inputs would be useful but it probably deserves a separate discussion.